src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversions.cs

```diff
@@ -709,14 +709,9 @@ protected override bool HasImplicitTupleConversion(BoundExpression source, TypeS
             var tupleExpression = (BoundTupleLiteral)source;
             var arguments = tupleExpression.Arguments;
 
-            // unwrap tuple to its underlying
-            if (destination.IsTupleType)
-            {
-                destination = ((TupleTypeSymbol)destination).UnderlyingTupleType;
-            }
-
-            // check if underlying type is actually a possible underlying type for a tuple of given arity
-            if(!Compilation.IsWellKnownTupleType(destination, arguments.Length))
+            // check if the type is actually compatible type for a tuple of given cardinality
+            int cardinality;
+            if (!destination.IsTupleCompatible(out cardinality) || cardinality != arguments.Length)
```

Comment:
This seems to be a very common check. Should it be a helper? 
And you would not need to declare a temp for cardinality